src/components/admin/InviteAdminModal/InviteAdminModal.tsx

```diff
@@ -27,29 +43,84 @@ export const InviteAdminModal: React.FC<InviteAdminModalProps> = ({
     []
   );
 
-  const { register } = useForm({
+  const { register, watch } = useForm({
     reValidateMode: "onChange",
     defaultValues,
   });
 
-  const addAdmin = async () => {
+  const searchQuery = watch("userName");
+
+  const { value: algoliaValue } = useAlgoliaSearch(searchQuery);
+
+  const foundUsers = algoliaValue?.[AlgoliaSearchIndex.USERS].hits;
+
+  const [{ loading: isAddingAdmins }, addNewAdmins] = useAsyncFn(async () => {
     if (!worldId) {
       return;
     }
 
-    // await addWorldAdmin(worldId, "userId placeholder")
+    const userIds = selectedUsers.map((user) => user.objectID) as UserId[];
+
+    await addWorldAdmins(worldId, userIds);
     onHide();
+  });
+
+  const selectUser = (user: User) => {
+    setSelectedUsers((selectedUsers) => {
+      if (selectedUsers.includes(user)) {
+        const newUsers = selectedUsers.filter(
+          (selectedUser) => selectedUser !== user
+        );
+        return newUsers;
+      }
+
+      return [...selectedUsers, user];
+    });
   };
 
+  const checkAdminAccess = (user: User) => selectedUsers.includes(user);
+
+  const hasFoundUsers = !!foundUsers?.length;
+  const hasSelectedUsers = selectedUsers?.length;
+
   return (
     <Modal show={show} onHide={onHide} autoHide>
-      <Input placeholder="Search users" name="userName" register={register} />
+      {hasFoundUsers && (
+        <div className="flex justify-center">{`${foundUsers.length} users found`}</div>
```

Comment:
[Q] Should we bother with singular/plural here?

Reply:
Yup, I'll improve the wording.